index.js

```diff
@@ -371,9 +371,18 @@ module.exports = function (log, indexesPath) {
     newIndexes[indexName].bitset.add(seq)
   }
 
+  // concurrent index update
+  const waitingIndexUpdate = {}
```

Comment:
`Map` is more suitable (also for performance) if you're adding and `delete`ing frequently